ops/internal/manage/depsets.go

```diff
@@ -0,0 +1,243 @@
+package manage
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/config"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	errInvalidActivationTime = errors.New("activation_time is before interop_time")
+	errDepsetLengths         = errors.New("inconsistent depset lengths")
+	errInconsistentDepsets   = errors.New("inconsistent depset values")
+	errMissingAddress        = errors.New("missing address")
+	errDuplicateChainIndex   = errors.New("duplicate chain index")
+)
+
+type DepsetChecker struct {
+	lgr             log.Logger
+	addrs           config.AddressesJSON
+	diskChainCfgs   map[uint64]DiskChainConfig
+	processedChains map[uint64]bool
+	chainsProcessed int
+}
+
+func NewDepsetChecker(logger log.Logger, cfgs []DiskChainConfig, addrs config.AddressesJSON) (*DepsetChecker, error) {
+	dc := &DepsetChecker{
+		lgr:             logger,
+		addrs:           addrs,
+		diskChainCfgs:   make(map[uint64]DiskChainConfig),
+		processedChains: make(map[uint64]bool),
+	}
+
+	for _, cfg := range cfgs {
+		dc.diskChainCfgs[cfg.Config.ChainID] = cfg
+	}
+
+	logger.Info("loaded depset checker instance with chain configs", "numDiskCfgs", len(cfgs))
+	return dc, nil
+}
+
+func (dc *DepsetChecker) Check() error {
+	for _, cfg := range dc.diskChainCfgs {
+		if dc.processedChains[cfg.Config.ChainID] {
+			// already processed this chain
+			continue
+		}
```

Comment:
Nit. This worried me when I fast saw it, because I thought it meant we wouldn't revisit a chain at all once we passed over it once. Digging in further, I think we are fine, but might be nice to explain why here. 
```suggestion
		if dc.processedChains[cfg.Config.ChainID] {
			// already processed this chain (it may be revisited anyway during checkOffChain)
			continue
		}
```

For reference, here is the test case I was worried about 98a1d0fda815f6e8173e57c751bf840d5c24022b. Happily the checker does report an error.